File: python/0.0-Basics/0.3-build-up-logical-thinking/0.3.09-pattern.py
from utils.clear import clear


def pattern_9():
    row = 5
    column = 10

    for i in range(1, row + 1):
        for j in range(1, column):

            if j <= column / 2:
                if j >= i:
                    print("*", end="")
                elif j < i:
                    print(" ", end="")
            elif j > column / 2:
                key = column - j
                if key >= i:
                    print("*", end="")
                elif key < i:
                    print(" ", end="")

        print(" ",i)

clear()
pattern_9()

File: js/src/0.0-Basics/0.3-build-up-logical-thinking/0.3.09-pattern.js
function log(val) {
    process.stdout.write(String(val));
}

function patter_9() {
    let row = 5;
    let column = 10;

    for (let y = 1; y < row + 1; y++) {
        for (let x = 1; x < column; x++) {
            if (x <= column - column / 2) {
                if (x >= y) {
                    log('*');
                } else {
                    log(' ');
                }
            } else {
                let key = column - x;

                if (key >= y) {
                    log('*');
                } else {
                    log(' ');
                }
            }
        }
        console.log();
    }
}

console.clear();
patter_9();
